### ISSUE TITLE
Problem displaying string values

### ISSUE THREAD

**Issue by gilesknap** [TNThieding/exif#12]
I am unable to display string values e.g.
[example image](https://photos.app.goo.gl/NyjS8yuoQEce24iR9) 
```
with open('/home/hgv27681/Downloads/IMG_20190317_104705.jpg', 'rb') as image_file:
                ex = exif.Image(image_file)
ex.image_description
Traceback (most recent call last):
  File "<input>", line 3, in <module>
  File "/home/hgv27681/.local/share/virtualenvs/exif-0yvCzAKq/lib/python3.7/site-packages/exif/_image.py", line 57, in __getattr__
    return getattr(self._segments['APP1'], item)
  File "/home/hgv27681/.local/share/virtualenvs/exif-0yvCzAKq/lib/python3.7/site-packages/exif/_app1_metadata.py", line 205, in __getattr__
    return ifd_tag.read()
  File "/home/hgv27681/.local/share/virtualenvs/exif-0yvCzAKq/lib/python3.7/site-packages/exif/_ifd_tag.py", line 143, in read
    retval = self._read_as_ascii()
  File "/home/hgv27681/.local/share/virtualenvs/exif-0yvCzAKq/lib/python3.7/site-packages/exif/_ifd_tag.py", line 155, in _read_as_ascii
    current_ascii_val = self.parent_segment_hex.read(cursor, 1)
  File "/home/hgv27681/.local/share/virtualenvs/exif-0yvCzAKq/lib/python3.7/site-packages/exif/_hex_interface.py", line 101, in read
    raise ValueError("byte offset exceeds length of segment")
ValueError: byte offset exceeds length of segment
```

**Comment by aushev** [comment]
I have the same issue with other attributes, like `gps_latitude_ref`, `gps_longitude_ref` etc. - they are all listed as existing when calling `dir(image)` but throw `ValueError: byte offset exceeds length of segment` error. Can't use this package because of the error.

**Comment by msm1089** [comment]
Could this be related to #24 ?
Looks like the `..._ref` attributes are shown as xx00, xx01 etc. Maybe @TNThieding has made some changes/fixes that have went wrong.
Would be great to get this working again as this is the easiest package to use for EXIF!

**Comment by TNThieding** [comment]
I just released version 0.8.3 of the package (merge request [here](https://gitlab.com/TNThieding/exif/merge_requests/24)) which corrects this issue. The underlying problem was that strings of 3 characters or less (e.g., "dav" in Giles' case) are stored directly in the EXIF tag data structure rather than storing a pointer to another address in the file with the ASCII value. The package now handles this case of having string of 3 characters or less. My apologies for how long it took to address this issue! Don't hesitate to follow-up with an additional issue if necessary.

**Comment by gilesknap** [comment]
Thanks!
